src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesNodeCommandGroupHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    [ProjectCommand(VSConstants.CMDSETID.StandardCommandSet2K_string, (long)VSConstants.VSStd2KCmdID.QUICKOBJECTSEARCH)]
+    [AppliesTo(ProjectCapability.PackageReferences)]
+    [Order(ProjectSystem.Order.Default)]
+    internal sealed class SuppressObjectBrowserForPackageReferenceCommand : AbstractSingleNodeProjectCommand
+    {
+        private static readonly Task<CommandStatusResult> s_suppressedResult = Task.FromResult(new CommandStatusResult(handled: true, null, CommandStatus.NotSupported | CommandStatus.Invisible));
+        
+        protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string? commandText, CommandStatus progressiveStatus)
+        {
+            if (node.Flags.Contains(DependencyTreeFlags.NuGetPackageDependency))
+            {
+                // Suppress "Open in Object Browser" for NuGet packages
+                return s_suppressedResult;
```

Comment:
Can you add another property to `GetCommandStatusResult` called `NotSupported`?

Reply:
This isn't just `NotSupported`, it's also `Invisible`. Both are important, to ensure CPS doesn't show the item.

Reply:
There's 4 different states:

- I don't handle the command
- I do handle the command
- The command is disabled
- The command is disabled and hidden

We have states for the first two, we can come up with a name that matches the last one.

Reply:
In 12ed9d8f09f783850238f5ffb13ab116338c300e I went with `GetCommandStatusResult.Suppressed`.